js/pokemon.js

```diff
@@ -357,17 +370,49 @@ function revealPokemon(correctlyGuessed, language) {
 
 function generateNewNumbers(force) {
 
-    if(force || (currentGen !== newGen)) {
-        var i=0, j;
-        
+    if(force || !sortCompareArrays(currentGen, newGen)) {
+
+        console.log("generating New Pokemon array")
+
         upcomingPokemon = new Array();
         upcomingPokemonArrayPos = 0;
-        
-        for(j=minPokemonNumber; j<=maxPokemonNumber; j++) {
-            upcomingPokemon[i] = j;
-            i++;
+        var i = 0;
+
+        for (var j=0; j < newGen.length; j++) {
```

Comment:
On a wider conversation, take a look at http://underscorejs.org/

Seriously powerful library, makes a lot of collection handling much easier. If you refactored your code with that you could do a lot with groupings (group by generation) and shuffling for example. 2 lines of code, very powerful and makes everything so much easier to read.